src/overlays/actors/ovl_En_Dha/z_en_dha.c

```diff
@@ -27,31 +40,364 @@ const ActorInit En_Dha_InitVars = {
+        this->unk_1D0.y = 0;
+        this->unk_1D0.z = this->unk_1D0.y;
+        this->unk_1D6.x = this->unk_1D0.y;
+        if (Math_Vec3f_DistXYZ(&playerPos, &this->unk_1DC) <= 12.0f) {
+            if (this->unk_1CC == 0) {
+                if (globalCtx->unk_11D4C(globalCtx, player) != 0) {
+                    this->unk_1CA = 0;
+                    this->unk_1CC++;
+                    if (this->actor.attachedA != NULL) {
+                        this->actor.attachedA->params = 1;
+                    }
+                    Audio_PlayActorSound2(&this->actor, NA_SE_EN_DEADHAND_GRIP);
+                }
+            } else {
+                this->unk_1CA += 0x1194;
+                this->unk_1D0.y = (Math_Sins(this->unk_1CA) * 1820.0f);
+                if ((player->stateFlags2 & 0x80) == 0) {
+                    this->unk_1CC = 0;
+                    func_809ECF60(this);
+                    return;
+                }
+                if (this->unk_1CA < -0x6E6B) {
+                    Audio_PlayActorSound2(&this->actor, NA_SE_EN_DEADHAND_GRIP);
+                }
+            }
+            func_80035844(&this->unk_1E8, &playerPos, &this->unk_1D0.z, 0);
+            this->unk_1D6.x = ((this->unk_1D6.x - this->actor.shape.rot.y) - this->unk_1D0.y);
+            this->unk_1D0.z = (((this->unk_1D0.z - this->actor.shape.rot.x) - this->unk_1CE) - this->unk_1D0.x);
+        } else {
+            if ((player->stateFlags2 & 0x80) != 0) {
+                if (&this->actor == player->actor.attachedA) {
+                    player->stateFlags2 &= ~0x80;
+                    player->actor.attachedA = NULL;
+                    player->unk_850 = 200;
+                }
+            }
+            if (this->actor.initPosRot.rot.z != 0) {
+                Audio_PlayActorSound2(&this->actor, NA_SE_EN_DEADHAND_HAND_AT);
+                this->actor.initPosRot.rot.z = 0;
+            }
+        }
+        this->actor.shape.rot.y = Math_Vec3f_Yaw(&this->actor.posRot, &playerPos);
+        Math_SmoothScaleMaxMinF(&this->unk_1DC, playerPos.x, 1.0f, 16.0f, 0.0f);
+        Math_SmoothScaleMaxMinF(&this->unk_1DC.y, playerPos.y, 1.0f, 16.0f, 0.0f);
+        Math_SmoothScaleMaxMinF(&this->unk_1DC.z, playerPos.z, 1.0f, 16.0f, 0.0f);
+        func_80035844(&this->unk_1F4, &this->unk_1DC, &unkVar3, 0);
+        Matrix_Translate(this->unk_1DC.x, this->unk_1DC.y, this->unk_1DC.z, 0);
+        Matrix_RotateRPY(unkVar3, unkVar2, 0, 1);
+        Matrix_MultVec3f(&D_809ED758, &this->unk_1F4);
+        Matrix_Translate(this->actor.posRot.pos.x, this->actor.posRot.pos.y, this->actor.posRot.pos.z, 0);
+        func_80035844(&this->actor.posRot, &this->unk_1F4, &unkVar3, 0);
+        Matrix_RotateRPY(unkVar3, unkVar2, 0, 1);
+        Matrix_MultVec3f(&D_809ED74C, &this->unk_1F4);
+        this->unk_1CE = Math_Vec3f_Pitch(&this->actor.posRot, &this->unk_1F4);
+        result = ((Math_Vec3f_Yaw(&this->actor.posRot, &this->unk_1F4) - this->actor.shape.rot.y) << 0x10) >> 0x10;
```

Comment:
The ` << 0x10) >> 0x10` there is due to a s16 cast, the shifts shouldn't appear in the C code.